starknet/rust/src/lib.rs

```diff
@@ -34,6 +35,7 @@ pub extern "C" fn freeCstr(ptr: *mut c_char) {
         if ptr.is_null() {
             return;
         }
+        // is it force destruction to deallocate memory?
         let _ = CString::from_raw(ptr);
```

Comment:
Yeap, exactly. `CString`takes the ownership of the pointer and releases it as it goes out of scope.